app/services/alerts/generate_subscription_events.rb

```diff
@@ -0,0 +1,47 @@
+module Alerts
+  class GenerateSubscriptionEvents
+    def initialize(school)
+      @school = school
+    end
+
+    def perform
+      @school.alerts.latest.each do |alert|
+        process_alert(alert)
+      end
+    end
+
+  private
+
+    def process_alert(alert)
+      if any_subscriptions?(@school, alert.alert_type)
```

Comment:
Not sure this guard is necessary. If there are no subscriptions the `each` block in process subscriptions won't run which is effectively the same thing?

Reply:
Yep - good spot